template/koa-rest-api/src/framework/validation.ts

```diff
@@ -35,41 +35,38 @@ type InvalidFields = Record<string, string>;
  * @see [union error example](./validation.test.ts)
  */
 const parseInvalidFieldsFromError = (err: z.ZodError): InvalidFields =>
-  Object.fromEntries(parseTuples(err, {}));
+  Object.fromEntries(parseTuples(err.issues));
 
 const parseTuples = (
-  { errors }: z.ZodError,
-  unions: Record<number, number[]>,
+  errors: core.$ZodIssue[],
+  basePath: Array<string | number | symbol> = [],
+  unions: Record<number, number[]> = {},
 ): Array<readonly [string, string]> =>
   errors.flatMap((issue) => {
-    if (issue.code === ZodIssueCode.invalid_union) {
```

Comment:
Accessing `ZodIssueCode` is marked as deprecated. The comments say to check directly against the string